### DISCUSSION TITLE
RPM release

### DISCUSSION THREAD

**Discussion by danielpodwysocki** [AcademySoftwareFoundation/OpenRV#76] [Ideas]
Hello,

Is an RPM release in the plans for this ? It would be great to have one to easily get this deployed.

**Comment by khw7096** [comment]
Good idea. :-)

**Comment by geffrak** [comment]
Hi @danielpodwysocki @khw7096,

I'm curious to know why you want an rpm package. We aim to have an RV archive that contains all the dependencies required to run it, so we can only unzip/untar and run.

The current commercial version doesn't have an RPM, and it's not something that causes issues.

**Comment by rogernelson** [comment]
Hi! If you're asking about that because you'd like to build it on one machine and distribute it throughout the facility, you could potentially zip/tar the installation directory and use that.

**Comment by lgritz** [comment]
If it's got a cmake build system now, it should be straightforward to augment it with cpack instructions which can make rpm (and many other package formats) out of the built results. Adding that would, I'm sure, make a great contribution to the project, and it's just build script changes and shouldn't require any knowledge of the rv internals.

(The mechanics of how if or how the ORI org might build and distribute binaries are a separate topic. I'm just commenting on the topic of "how can one build an rpm that can then be distributed.")

**Comment by danielpodwysocki** [comment]
Hi @geffrak ,

For most medium to big studios I think it's not as relevant and that's why it hasn't come up for the commercial version. We all have own deployment workflows this can fit into.

However, now that this is open source, individual users and smaller houses wanting to try it out would benefit greatly - running a couple yum commands is way more straightforward than having to deal with this on their own.

**Comment by danielpodwysocki** [comment]
@lgritz GitHub supports automated releases - there are other options, but this one would be the most "plug&play" and not require anything extra.

**Comment by geffrak** [comment]
I understand the need. It is not something available right now because the focus was to put the code public.

The goal is to eventually have some CI/CD in place (because we can’t use Autodesk’s one) and maybe it will gives us installable artifacts. 

Thank you for your input!

**Comment by bernie-laberge** [comment]
I am concluding this discussion. However, I have created the following issue to ensure that the proposed enhancement is implemented by the community:
https://github.com/AcademySoftwareFoundation/OpenRV/issues/584
